### DISCUSSION TITLE
Chat tabs and other player chat improvements

### DISCUSSION THREAD

**Discussion by Evan-R-Rogers** [unitystation/unitystation#8457] [Ideas]
Looking to improve the overall chat feature for players in-game.

### Suggestions:

**Different tabs for different chats** - Firstly, I believe that the different chats should be split up into different tabs, possibly with notifications. This would make it easier to read OOC as its not cluttered with messages from the world or other players. I would propose a way to make this customizable per client but it may be too early for that.

**Timestamps** - Self explanatory. Keep it basic yet easy to read. Only for OOC messages.

**Admin messages** - Have messages from admins a different color (As well as bolded?), to make their messages easier to distinguish. Would also help prevent admin impersonators from harassing players.

**Random fixes** -  The scroll bar seems to go passed where the true chat window stops, I suggest allowing text all the way to the top of the screen nearly. Furthermore, the scrollbar is cut off at the top, going off-screen. Large messages, mainly those that detail objects being destroyed, completely disappear if they become too large. Special chat colors? Special chat fonts? Add a dark blue, transparent background to the chat window as it's often hard to read (Make this an option somewhere). Bring back message stacking. Players that are labelled as 'Unknown' should of course be labelled as such when emoting in chat.

**Comment by Fr-Dae** [comment]
like here ?
https://github.com/unitystation/unitystation/issues/6549

**Reply by Fr-Dae** [reply]
for the time, local ? global ? server ?

because the server time is reset between each round.
the global time is determined by the host
and the local time changes (for example here gmt+1)
